### PR TITLE
[auth] Implement Reset Password functionality + build frontend screen 

### PR DESCRIPTION
**Pull request by allisonhongberkeley**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

In this PR, I completed the frontend for the "Reset Password" screen and also implemented the password-reset functionality where clicking "send email" will generate a link to reset the password and email it to the user. 

In addition, if the user inputs an incorrect email, an error message will appear. If the email validation is sent successfully, a success message will appear. 

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

Screen Shot 2023-02-18 at 4 02 32 PM
Screen Shot 2023-03-02 at 6 50 53 PM
Screen Shot 2023-03-02 at 6 49 42 PM
Screen Shot 2023-03-02 at 6 51 20 PM

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

Go to the login page, click Forgot Password, and try sending yourself a reset email. 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Things that still don't work: 
[NA]

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: src/screens/auth/ForgotPassword.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        </HeadingContainer>
+
+        <RowContainer>
+          <Body1Text>
+            Enter the email associated with your account, and we will send an
+            email with instructions to reset your password.
+          </Body1Text>
+        </RowContainer>
+
+        <Body1Text style={Styles.bold}>Email</Body1Text>
```
Comment (sauhardjain) [inline src/screens/auth/ForgotPassword.tsx]: [nit] let's not use in-line styling unless absolutely necessary. 

also styling slightly differs from Figma mockup — add spacing on top of "Email" and use bold text.

File: src/screens/auth/ForgotPassword.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        <InputField
+          onChange={onChangeEmail}
+          value={email}
+          placeholder="Enter email"
+        />
+        {showErrorMessage && errorMessage && (
+          <Body1Text style={Styles.errorText}>{errorMessage}</Body1Text>
```
Comment (sauhardjain) [inline src/screens/auth/ForgotPassword.tsx]: let's incorporate an error message here instead of using the Firebase Auth error. 

https://www.figma.com/file/XcRTx9VXlBGILye8UsKCnH?node-id=1705:5088#374429495

File: src/screens/auth/LoginScreen.tsx [hunk 3]
```diff
@@ -50,6 +53,7 @@ export default function LoginScreen() {
   // TODO: implement password reset functionality @selene-huang
```
Comment (sauhardjain) [inline src/screens/auth/LoginScreen.tsx]: can you remove this TODO?

File: src/utils/authUtils.ts [hunk 4]
```diff
@@ -16,6 +17,23 @@ export const setAuthErrorMessage = (
...
 
+export const forgotPassword = async (
+  dispatch: AuthDispatch,
+  params: { email: string },
+) => {
+  const auth = getAuth();
+  await sendPasswordResetEmail(auth, params.email)
+    .then(() => {
+      // eslint-disable-next-line no-console
+      console.log('Email Sent!');
```
Comment (sauhardjain) [inline src/utils/authUtils.ts]: let's get rid of the console log and add a success message when the email is successfully sent!

**Review by sauhardjain — CHANGES_REQUESTED** [review]
GREAT job on this sprint Allison! I've left a few comments that shouldn't take too long to fix. Apart from these, other next steps for this sprint include:

- [x] Customize the password reset email to include "Vouchers 4 Veggies" instead of "project-405408181772"
- [x] Display a success message of "Email sent successfully" after the email is sent.
- [x] Display an error message when an incorrect email is entered. (Also mentioned in one of the comments)

File: src/components/common/StandardLogo 2.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import v4vLogo from '../../../assets/top-bar-logo.png';
+
+const LogoContainer = styled.ImageBackground`
+  width: 60px;
+  height: 70px;
+`;
+
+export default function StandardLogo() {
+  return <LogoContainer source={v4vLogo} />;
+}
```
Comment (sauhardjain) [inline src/components/common/StandardLogo 2.tsx]: This file shouldn't exist @allisonhongberkeley! Make sure you delete this and any other duplicate files.

File: src/screens/scanning/ManualVoucherScreen 2.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
```
Comment (sauhardjain) [inline src/screens/scanning/ManualVoucherScreen 2.tsx]: Duplicate file.

**Review by sauhardjain — COMMENTED** [review]
Noticed that your latest commits contain two duplicate files — make sure you discard these. As far as I know, this issue is related to syncing your files to iCloud. A quick fix is to place your local repo in a folder not synced to iCloud (you can also force this by adding .tmp or .nosync to any folder name).

Apart from this, review my previous comments as well so we can get the PR merged ASAP.

File: src/screens/auth/ForgotPassword.tsx (hunk 1 above)
Comment (allisonhongberkeley) [reply src/screens/auth/ForgotPassword.tsx]: fixed

File: src/utils/authUtils.ts (hunk 4 above)
Comment (allisonhongberkeley) [reply src/utils/authUtils.ts]: removed

File: src/screens/scanning/ManualVoucherScreen 2.tsx (hunk 6 above)
Comment (allisonhongberkeley) [reply src/screens/scanning/ManualVoucherScreen 2.tsx]: deleted

File: src/screens/auth/styles.ts [hunk 7]
```diff
@@ -104,4 +108,10 @@ export const Styles = StyleSheet.create({
   errorText: {
     color: Colors.alertRed,
   },
+  successText: {
+    color: Colors.alertGreen,
+  },
+  bottomMargin: {
+    marginBottom: 20,
+  },
```
Comment (sauhardjain) [inline src/screens/auth/styles.ts]: Are you using these styles?

File: assets/Fonts.tsx [hunk 8]
```diff
@@ -53,6 +53,18 @@ export const MagentaText = styled.Text`
...
 
+export const ErrorText = styled.Text`
+  color: ${Colors.alertRed};
+`;
+
+export const SuccessText = styled.Text`
+  color: ${Colors.alertGreen};
+`;
+
+export const Bold = styled.Text`
```
Comment (sauhardjain) [inline assets/Fonts.tsx]: Rename this to BoldText to be consistent with the font naming convention.

File: src/screens/auth/ForgotPassword.tsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+  };
+
+  return (
+    <LoginContainer>
+      {/* logo placeholder */}
+      <LogoContainer>
+        <View style={{ backgroundColor: 'black', width: 50, height: 59.29 }}>
+          <Text style={{ color: 'white' }}>{`\n  Logo`}</Text>
+        </View>
+      </LogoContainer>
```
Comment (sauhardjain) [inline src/screens/auth/ForgotPassword.tsx]: Use the StandardLogo component here to actually implement the V4V logo in the screen.

File: src/screens/auth/ForgotPassword.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+  HeadingContainer,
+  LoginContainer,
+  LogoContainer,
+  VerticalSpacingButtonContainer,
+  WhiteText,
+  RowContainer,
+  BottomMargin,
+} from './styles';
+
+export default function ForgotPasswordScreen() {
```
Comment (sauhardjain) [inline src/screens/auth/ForgotPassword.tsx]: Update the file name to `screens\auth\ForgotPasswordScreen.tsx` to match the file naming convention for screens.

File: src/screens/auth/ForgotPassword.tsx [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        {showErrorMessage && errorMessage && (
+          <Body1Text>
+            <ErrorText>We could not find that email address!</ErrorText>
+          </Body1Text>
+        )}
+        {showSuccessMessage && successMessage && (
+          <Body1Text>
+            <SuccessText>Email sent! Check your inbox to reset.</SuccessText>
+          </Body1Text>
+        )}
```
Comment (sauhardjain) [inline src/screens/auth/ForgotPassword.tsx]: I've been getting this error as soon as I try to navigate to the ForgotPassword screen and traced the issue to these conditional operators. You should use ternary operators instead because we might run into type errors otherwise. 

See this for more info: https://stackoverflow.com/questions/52368342/invariant-violation-text-strings-must-be-rendered-within-a-text-component

IMG_13D2A7171CB2-1

**Review by sauhardjain — CHANGES_REQUESTED** [review]

**Comment by sauhardjain** [comment]
Replaced by newer PR with cleaner commit history.
